app/src/main/java/com/konarsubhojit/synckro/data/worker/SyncWorker.kt

```
@@ -0,0 +1,173 @@
+                is SyncEngine.Result.Terminal -> {
+                    // Misconfigured pair (e.g. revoked auth, unsupported provider).
+                    // Don't retry forever — cancel the periodic chain so the
+                    // user can re-authenticate / reconfigure the pair.
+                    Timber.w("Terminal sync failure for pair %d: %s", pairId, r.reason)
+                    WorkManager.getInstance(applicationContext).cancelUniqueWork(uniqueName(pairId))
+                    Result.failure()
+                }
+            }
+        } catch (c: CancellationException) {
+            throw c
+        } catch (t: Throwable) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)